tests/test_motors.py

```diff
@@ -18,55 +30,67 @@
 import numpy as np
 import pytest
 
-from lerobot import available_robots
+from lerobot import available_motors
 from lerobot.common.robot_devices.motors.utils import MotorsBus
-from lerobot.common.robot_devices.robots.factory import make_robot
 from lerobot.common.robot_devices.utils import RobotDeviceAlreadyConnectedError, RobotDeviceNotConnectedError
-from lerobot.common.utils.utils import init_hydra_config
-from tests.utils import ROBOT_CONFIG_PATH_TEMPLATE, require_robot
+from tests.utils import TEST_MOTOR_TYPES, require_motor
+
+DYNAMIXEL_PORT = "/dev/tty.usbmodem575E0032081"
+DYNAMIXEL_MOTORS = {
+    "shoulder_pan": [1, "xl430-w250"],
+    "shoulder_lift": [2, "xl430-w250"],
+    "elbow_flex": [3, "xl330-m288"],
+    "wrist_flex": [4, "xl330-m288"],
+    "wrist_roll": [5, "xl330-m288"],
+    "gripper": [6, "xl330-m288"],
+}
+
+
+def make_motors_bus(motor_type: str, **kwargs) -> MotorsBus:
+    if motor_type == "dynamixel":
+        from lerobot.common.robot_devices.motors.dynamixel import DynamixelMotorsBus
 
+        port = kwargs.pop("port", DYNAMIXEL_PORT)
+        motors = kwargs.pop("motors", DYNAMIXEL_MOTORS)
+        return DynamixelMotorsBus(port, motors, **kwargs)
 
-def make_motors_bus(robot_type: str) -> MotorsBus:
-    # Instantiate a robot and return one of its leader arms
-    config_path = ROBOT_CONFIG_PATH_TEMPLATE.format(robot=robot_type)
-    robot_cfg = init_hydra_config(config_path)
-    robot = make_robot(robot_cfg)
-    first_bus_name = list(robot.leader_arms.keys())[0]
-    motors_bus = robot.leader_arms[first_bus_name]
-    return motors_bus
+    else:
+        raise ValueError(f"The motor type '{motor_type}' is not valid.")
 
 
-@pytest.mark.parametrize("robot_type", available_robots)
-@require_robot
-def test_find_port(request, robot_type):
+# TODO(rcadene): implement mocked version of this test
```

Comment:
I'm wondering if there is a way where instead of having separate tests for mock and not mock motors, etc. it is one test and then within the test differentiate (for example), this way we are always using the same tests.



```
def test_find_port(request, motor_type):
    if motor_type == "dynamixel":
        from lerobot.common.robot_devices.motors.dynamixel import find_port
        find_port()
    elif motor_type == "mock_dynamixel":
        pass  # Replace with mock behavior or an assertion if necessary```

Reply:
We could do that, but it's much more time consuming for us.
I would prefer to design my unit tests once, just for real cameras, then mock the cameras.

Also using `monkeypatch` upstream allows to write a minimal amount of code.
We couldnt `monkeypatch` upstream , but dynamically in the tests, which increase the complexity and maintainability burden .

Reply:
Got it! Sounds good!